src/components/CreateCourse/CreateCourse.tsx

```diff
@@ -158,16 +204,23 @@ export const CreateCourse: React.FC = () => {
 			duration: Number(duration),
 			authors: courseAuthorList.map((courseAuthor: Author) => courseAuthor.id),
 		};
+
 		if (isNewCourseValid(newCourse)) {
+			const newAuthorList = [];
 			courseAuthorList
-				.filter((courseAuthor) => !authorList.includes(courseAuthor))
+				.filter((courseAuthor) => !authors.includes(courseAuthor))
 				.forEach((author) => {
-					authorList.push(author);
+					newAuthorList.push(author);
 				});
+
 			authorList
-				.filter((author) => !mockedAuthorsList.includes(author))
-				.forEach((newAuthor) => mockedAuthorsList.push(newAuthor));
-			mockedCoursesList.push(newCourse);
+				.filter((author) => !authors.includes(author))
+				.forEach((author) => newAuthorList.push(author));
+
+			newAuthorList.forEach((author) => dispatch(AddAuthorAction(author)));
```

Comment:
maybe better to create an action that will add all authors, and logic by adding this authors hide in reducer?)